build/Templating.targets

```diff
@@ -0,0 +1,68 @@
+<Project>
+  <PropertyGroup>
+    <TemplatingProjectRoot>$(MSBuildThisFileDirectory)..\modules\Templating\</TemplatingProjectRoot>
+  </PropertyGroup>
+
+  <Target Name="BuildTemplates" DependsOnTargets="GeneratePropsFiles">
+    <PropertyGroup>
+      <GeneratedNoTimestampPackageVersionPropsPath>$(IntermediateDir)dependencies.notimestamp.props</GeneratedNoTimestampPackageVersionPropsPath>
+      <TemplateProjCommmonProperties>
+        RepositoryRoot=$(TemplatingProjectRoot);
+        DotNetRestoreSourcesPropsPath=$(GeneratedRestoreSourcesPropsPath);
+        BuildNumber=$(BuildNumber);
+        Configuration=$(Configuration);
+      </TemplateProjCommmonProperties>
+      <TemplateProjProperties>
+        $(TemplateProjCommmonProperties);
+        DotNetPackageVersionPropsPath=$(GeneratedPackageVersionPropsPath);
+        VersionMetadata=timestamped;
+      </TemplateProjProperties>
+      <TemplateProjNoTimestampProperties>
+        $(TemplateProjCommmonProperties);
+        DotNetPackageVersionPropsPath=$(GeneratedNoTimestampPackageVersionPropsPath);
+        VersionMetadata=notimestamp;
+      </TemplateProjNoTimestampProperties>
+    </PropertyGroup>
+
+    <!-- Produce regular, timestamped templates for pre-release builds -->
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
```

Comment:
For templates, I'm not sure we need to have this distinction between timestamped and no-timestamped package versions. Since end users don't install the template packages manually, there's no drawback to having the build numbers in the packages even in the final releases. In fact our current template packages do have build numbers in their names when shipped.

Would it make things much simpler for you if you didn't need to make that distinction?

Reply:
Yes, would love to simplify this if possible. I was under the impression the final nupkg we gave to the dotnet/CLI team needed to contain `content/*/*.csproj` files that had code like this
```xml
 <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.3" />
```
Is this not the case?

Reply:
Oh I see. Yes that is right so I think what you have here should stay.

I meant that the template packages themselves (but not the runtime dependencies for generated projects) would be OK to have timestamped versions for release.

Reply:
Thanks for clarifying.

The only possible problem I see with this is how the CLI consumes our packages. Do they pull from a myget feed? If so, while we were preparing the final build of CLI and had to reset a non-timestamped package version, it comes really tricky to invalid nuget caches with a new version of `Microsoft.AspNetCore.SpaTemplates/2.0.3`. Maybe something like this?

Microsoft.AspNetCore.SpaTemplates 2.0.3-rtm-build-123-final => contains templates with non-timestamped versions
Microsoft.AspNetCore.SpaTemplates 2.0.3-rtm-build123 => contains templates with prerelease labels